_63C/digitalwallet.md

```diff
@@ -0,0 +1,7 @@
+---
+question: C9
+title: Can a digital wallet be an IdP?
+---
+While OpenID Connect and SAML are the most common federated identity protocols in use today, the Digital Identity Guidelines allow any protocol that meets the requirements outlined in the system to act as a federation protocol. Similarly, while a web-hosted server is the most common deployment for an IdP, the Digital Identity Guidelines do not assume any particular deployment model for IdPs. 
+
+There has been interest in recent years in using applications to store cryptographically protected material on a user-carried device, a practice sometimes referred to as a digital wallet. Given that the definition of IdP does not assume a publicly deployed web service, a digital wallet could serve the function of an IdP as long as it fulfills all the requirements laid out in the guidelines. Namely, the wallet would need to create a signed assertion (which still needs to meet all the general assertion requirements) and somehow transfer that assertion to the RP over a network using either a front-channel or back-channel presentation mechanism. The RP in turn would need to be able to validate the assertion's signature and contents, as well as determine that the IdP is a trustworthy party for generating the assertion. There are numerous methods for establishing an IdP-RP trust relationship discussed in the Digital Identity Guidelines, and any of these could be applicable to trusting assertions from a digital wallet style application.
```

Comment:
I presume that the wallet itself needs to do the signing in order to make sure that the assertion has the correct audience, etc. Is that correct, or does it sign a pre-signed assertion (which I suppose could be encapsulated in another assertion with audience)? In any case, trust for a wallet held by an individual subscriber seems like a scaling challenge.

Reply:
Yes, the wallet would need to sign the assertion, audience restrict it, and do all of those other things that are spelled out in the guidelines as requirements for assertions. Usually a wallet is acting more like an attribute provider so it would need to go an extra step of having an assertion package to wrap them in.

Trust is definitely a challenge in the space, but that's not the challenge we're trying to solve here. It's akin to having a personal IDP and convincing RPs to accept it -- technologically possible, and past that it's up to the RP.

Reply:
If we're capitalizing Guidelines this is another one:

```suggestion
There has been interest in recent years in using applications to store cryptographically protected material on a user-carried device, a practice sometimes referred to as a digital wallet. Given that the definition of IdP does not assume a publicly deployed web service, a digital wallet could serve the function of an IdP as long as it fulfills all the requirements laid out in the Guidelines. Namely, the wallet would need to create a signed assertion (which still needs to meet all the general assertion requirements) and somehow transfer that assertion to the RP over a network using either a front-channel or back-channel presentation mechanism. The RP in turn would need to be able to validate the assertion's signature and contents, as well as determine that the IdP is a trustworthy party for generating the assertion. There are numerous methods for establishing an IdP-RP trust relationship discussed in the Digital Identity Guidelines, and any of these could be applicable to trusting assertions from a digital wallet style application.
```

Reply:
Regarding the requirement to sign the assertion and related attribute information, what entity is the signing authority? I have looked at this as being self-asserted information -- so if the wallet owner is the signing authority then this is just signed self-asserted information. The principle in 800-63 is that self-asserted information does not bear trust. This is the question that I want to discuss on this FAQ.